frontend/src/pages/Results/index.tsx

```diff
@@ -0,0 +1,140 @@
+import { Category, Choice, Question } from '../../actions';
+import { useEffect, useState } from 'react';
+import { config } from '../../actions/config';
+import { SessionData, UserProgress, Result } from '../../actions/types';
+import axios from 'axios';
+import { connect } from 'react-redux';
+
+interface Props {
+  category: Category;
+  questions: Question[];
+  choices: Choice[][];
+  currentLogin: SessionData;
+}
+
+const ResultsPage = ({
+  currentLogin: { user, token },
+  category,
+  questions,
+  choices,
+}: Props) => {
+  const [score, setScore] = useState(0);
+  const [quizData, setQuizData] = useState<Result[]>();
+
+  useEffect(() => {
+    getUsersQuizData();
+  }, []);
+
+  useEffect(() => {
+    checkQuizStatus().then((res) => {
+      res.status === -1 && updateStatus(res.id);
+    });
+  }, [score]);
+
+  const updateStatus = (progressId: number) => {
+    axios.put(
+      `${config.URL}/users/${user.id}/${category.slug}/progress/${progressId}`,
+      { status: score },
+      {
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      }
+    );
+  };
+
+  const checkQuizStatus = async () => {
+    return axios
+      .get<UserProgress>(
+        `${config.URL}/users/${user.id}/${category.slug}/progress`,
+        {
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+        }
+      )
+      .then((res) => {
+        return res.data;
+      });
+  };
+
+  const getUsersQuizData = async () => {
+    await axios
+      .get<Result[]>(
+        `${config.URL}/users/${user.id}/${category.slug}/results`,
+        {
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+        }
+      )
+      .then((res) => {
+        setQuizData(res.data);
+      });
+  };
+
+  const getUsersAnswer = (index: number): string => {
+    if (!quizData) return '';
+    return choices[index].filter(
+      (result) => result.id === quizData[index].user_choice_id
+    )[0].choice;
+  };
+
+  const getCorrectAnswer = (index: number): Choice => {
+    return choices[index].filter((choice) => choice.is_correct)[0];
+  };
+
+  const getAnswerResult = (index: number): boolean => {
+    if (!quizData) return false;
+    return quizData[index].is_correct;
+  };
+
+  useEffect(() => {
+    getTotalScore();
+  }, [quizData]);
+
+  const getTotalScore = () => {
+    if (!quizData) return false;
```

Comment:
Since in line 93, you are not assigning it to a variable
```suggestion
    if (!quizData) return;
```